recommenders/models/ncf/dataset.py

```diff
@@ -1,321 +1,450 @@
+import os
+from collections import OrderedDict
 import random
 import numpy as np
 import pandas as pd
-import warnings
+import csv
+import logging
 
 from recommenders.utils.constants import (
     DEFAULT_ITEM_COL,
     DEFAULT_USER_COL,
-    DEFAULT_RATING_COL,
-    DEFAULT_TIMESTAMP_COL,
+    DEFAULT_RATING_COL
 )
 
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class DataFile():
+    """
+    DataFile class for NCF. Iterator to read data from a csv file.
+    Data must be sorted by user. Includes utilities for loading user data from
+    to file, formatting it and returning a Pandas dataframe.
+    """
+
+    def __init__(
+        self, filename, col_user, col_item, col_rating,
+        col_test_batch=None, binary=True
+    ):
+        self.filename = filename
+        self.col_user = col_user
+        self.col_item = col_item
+        self.col_rating = col_rating
+        self.col_test_batch = col_test_batch
+        self.expected_fields = [self.col_user, self.col_item, self.col_rating]
+        if self.col_test_batch is not None:
+            self.expected_fields.append(self.col_test_batch)
+        self.binary = binary
+        self.user2id, self.item2id, self.id2user, self.id2item = None, None, None, None
```

Comment:
id2user, id2item are not used in `DataFile`
Also no need to initialize the other two, since they are initialized in the following line.

Reply:
id2user and id2item are not used in DataFile but are included as a helper to make it easy to check a user/item against a given ID. I'll remove the unnecessary initialization.

Reply:
But this is in DataSet, not DataFile. In DataFile they are not intialized.